pkg/mcp/source/server_source_test.go

```diff
@@ -31,6 +31,25 @@ import (
 	"istio.io/istio/pkg/mcp/testing/monitoring"
 )
 
+type fakeRateLimiter struct {
```

Comment:
This could be shared between source/sink by placing it in the pkg/mcp/internal/test/ package.